docs/how-to-guides/dependabot-auto-merge.md

```diff
@@ -0,0 +1,117 @@
+# Dependabot with auto-merge
+
+[working-with-dependabot]: https://docs.github.com/en/code-security/dependabot/working-with-dependabot
+[automating-dependabot]: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions
+[configure-dependabot-yaml]: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
+[github-cli]: https://cli.github.com/
+
+[Dependabot][working-with-dependabot] is a security tool offered by GitHub.
+Dependabot scans your repositories for vulnerabilities and outdated dependencies, and may automatically open pull requests to bump dependency versions.
+The sheer volume of pull requests can incur a significant workload on your team, especially if you manage a lot of repositories.
+
+By completing this guide, Dependabot will automatically fix your insecure or outdated dependencies, and the changes will automatically get merged into your main branch.
+
+## Prerequisites
+
+* [GitHub command-line interface][github-cli] installed.
+
+## Enable Dependabot
+
+The contents of this file will depend on your project requirements. Do not use this file as-is.
+Please see [dependabot.yaml configuration syntax][configure-dependabot-yaml] for detailed instructions on how to configure Dependabot.
+
+!!! note ".github/dependabot.yaml"
+
+	```yaml
+	version: 2
+	updates:
+	  - die: &I didn't edit my config file
+	  - package-ecosystem: "github-actions"
+		directory: "/"
+		schedule:
+		  interval: "daily"
+		  time: "10:05"
+		  timezone: "Europe/Oslo"
+	  - package-ecosystem: "docker"
+		directory: "/"
+		schedule:
+		  interval: "daily"
+		  time: "10:05"
+		  timezone: "Europe/Oslo"
+	```
+
+## GitHub workflow for auto-merging Dependabot pull requests
+
+This workflow will trigger when dependabot opens a pull request.
+All minor and patch-level changes are automatically merged.
+Major version bumps needs manual merging.
+See also [Automating Dependabot with GitHub Actions][automating-dependabot].
+
+!!! note ".github/workflows/dependabot-auto-merge.yaml"
+
+	```yaml
+	name: Dependabot auto-merge
+	on: pull_request
+
+	permissions:
+	  contents: write
+	  pull-requests: write
+
+	jobs:
+	  dependabot:
+		runs-on: ubuntu-latest
+		if: ${{ github.actor == 'dependabot[bot]' }}
+		steps:
+		  - name: Dependabot metadata
+			id: metadata
+			uses: dependabot/fetch-metadata@v1
+			with:
+			  github-token: "${{ secrets.GITHUB_TOKEN }}"
+		  - name: Auto-merge changes from Dependabot
+			if: steps.metadata.outputs.update-type != 'version-update:semver-major' || steps.metadata.outputs.package-ecosystem == 'github_actions'
+			run: gh pr merge --auto --squash "$PR_URL"
+			env:
+			  PR_URL: ${{github.event.pull_request.html_url}}
+			  GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+	```
+
+## Enable branch protection and auto-merge on repository
```

Comment:
Skulle man også lenket til hvordan man gjør det i GUI-et?
Enable auto merge: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-auto-merge-for-pull-requests-in-your-repository
Branch protection: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule